airflow/models/taskinstance.py

```diff
@@ -1379,23 +1379,20 @@ def _run_raw_task(
 
             session.commit()
 
-    def _prepare_and_execute_task_with_callbacks(self, context, task):
+    def _execute_task_with_callbacks(self, context):
         """Prepare Task for Execution"""
         from airflow.models.renderedtifields import RenderedTaskInstanceFields
 
-        task_copy = task.prepare_for_execution()
```

Comment:
I can't easily parse the whole diff on my phone but we can't lose this call - it's important I think

Reply:
Did I just miss it getting moved?

Reply:
It's moved to the call up the stack

Reply:
and monday is perfecly fine :)

Reply:
From what I can tell, the most important part is this method is now calling things on the original `TI.task` instead of a copy of it. I never really understood why a copy was needed in the first place and will leave this for those who do…

Reply:
I believe this has a lot  to do with automated parameter passing via Xcom as there is some magic involved in "output" processing for TaskFlow.